src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/HACK_NodeHelper.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands.Ordering
+{
+    /// <summary>
+    /// These are helper functions to select items, expand and collapse folders in a IVsUIHierarchy.
+    /// This is hack due to CPS not exposing fuctionality to do this. They have it internally though.
+    /// </summary>
+    internal static class HACK_NodeHelper
```

Comment:
I would the HACK part of of this.  Instead, mention it as you do in the doc comment, and link to the bug where you asked for it to be made accessible (which may not get fixed, but would have discussion about why not).

Reply:
So, this might be able to go away soon. At this point in time, I just need Selecting a node. The reason is due to moving an item that you selected, it wouldn't remain selected.  No guarantees, but this https://devdiv.visualstudio.com/DevDiv/_workitems/edit/585954 may fix issue as CPS isn't doing the right thing when items were re-positioned.  // cc @jviau

Reply:
Let's make sure to rename before merging and add a comment with the issue link.

Reply:
I'm going to find out if we can make a slight change to CPS where we don't need this class anymore. I found where CPS does the logic for previously selected nodes.